lib/Search/ElasticSearch/ElasticSearchEngine.php

```diff
@@ -118,15 +119,39 @@ protected function validateQuery(SearchQuery &$query)
      */
     private function createSearchParams($query)
     {
-        $params = [
+        $searchStr = $query->getSearchString();
+
+        // Wildcard character required for Elasticsearch
+        $wildcardBe = "*";
+
+        // Override frontend wildcard character
+        if (isset($GLOBALS['sugar_config']['search_wildcard_char'])) {
+            $wildcardFe = $GLOBALS['sugar_config']['search_wildcard_char'];
+            if ($wildcardFe !== $wildcardBe && strlen($wildcardFe) === 1) {
+                $searchStr = str_replace($wildcardFe, $wildcardBe, $searchStr);
+            }
+        }
```

Comment:
@Dillon-Brown @mattlorimer Minor improvemenet. Maybe we could simplify the above by doing.

```php
        $wildcardFe = $GLOBALS['sugar_config']['search_wildcard_char'] ?? '';
        if ($wildcardFe !== $wildcardBe && strlen($wildcardFe) === 1) {
            $searchStr = str_replace($wildcardFe, $wildcardBe, $searchStr);
        }
```

Reply:
Same as above.

Reply:
thanks!